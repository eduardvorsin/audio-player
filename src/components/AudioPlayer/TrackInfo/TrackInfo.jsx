import PropTypes from 'prop-types';
import { memo } from 'react';
import { StyledTrackArtist, StyledTrackInfo, StyledTrackName } from "./StyledTrackInfo";

export const TrackInfo = memo(({
  trackName,
  trackArtist,
  className,
}) => {
  return (
    <StyledTrackInfo
      className={className}
      data-testid='track-info'
    >
      <StyledTrackName
        data-testid='track-name'
      >
        {trackName}
      </StyledTrackName>
      <StyledTrackArtist
        data-testid='track-artist'
      >
        {trackArtist}
      </StyledTrackArtist>
    </StyledTrackInfo>
  );
});

TrackInfo.propTypes = {
  trackName: PropTypes.string,
  trackArtist: PropTypes.string,
  className: PropTypes.string,
}

TrackInfo.defaultProps = {
  trackName: 'trackName',
  trackArtist: 'trackArtist',
  className: '',
}
