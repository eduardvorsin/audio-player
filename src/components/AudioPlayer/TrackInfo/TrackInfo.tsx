import { memo } from 'react';
import { StyledTrackArtist, StyledTrackInfo, StyledTrackName } from "./StyledTrackInfo";

interface TrackInfoProps {
  trackName: string,
  trackArtist: string,
  className?: string,
}

export const TrackInfo = memo<TrackInfoProps>(({
  trackName,
  trackArtist,
  className = '',
}) => {


  return (
    <StyledTrackInfo
      className={className}
      data-testid='track-info'
    >
      <StyledTrackName
        data-testid='track-name'
      >
        {trackName}
      </StyledTrackName>

      <StyledTrackArtist
        data-testid='track-artist'
      >
        {trackArtist}
      </StyledTrackArtist>
    </StyledTrackInfo>
  );
});