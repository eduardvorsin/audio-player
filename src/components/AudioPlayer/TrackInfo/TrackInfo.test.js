import { render, screen } from "@testing-library/react";
import { TrackInfo } from "./TrackInfo";

describe('Track Info component tests', () => {
  test('renders correctly', () => {
    render(<TrackInfo
      trackName='track'
      trackArtist='artist'
    />);

    expect(screen.getByText('track')).toBeInTheDocument();
    expect(screen.getByText('artist')).toBeInTheDocument();
  });

  test('renders without passed props', () => {
    render(<TrackInfo />);

    expect(screen.getByText('trackName')).toBeInTheDocument();
    expect(screen.getByText('trackArtist')).toBeInTheDocument();
  });

  test('className is assigned correctly', () => {
    render(
      <TrackInfo
        className='test'
      />
    );

    expect(screen.getByTestId('track-info')).toHaveClass('test');
  });

  test('snapshot with default prop values', () => {
    render(
      <TrackInfo />
    );

    expect(screen.getByTestId('track-info')).toMatchSnapshot();
  });

  test('snapshot with all the passed props', () => {
    render(
      <TrackInfo
        trackName='abc'
        trackArtist='ccb'
        className='test'
      />
    );

    expect(screen.getByTestId('track-info')).toMatchSnapshot();
  });
});
