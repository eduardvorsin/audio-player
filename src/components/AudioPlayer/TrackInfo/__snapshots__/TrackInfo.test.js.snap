// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track Info component tests snapshot with all the passed props 1`] = `
<div
  class="sc-bcXHqe kMOjqp test"
  data-testid="track-info"
>
  <p
    class="sc-gswNZR hQCPLi"
    data-testid="track-name"
  >
    abc
  </p>
  <p
    class="sc-dkrFOg"
    data-testid="track-artist"
  >
    ccb
  </p>
</div>
`;

exports[`Track Info component tests snapshot with default prop values 1`] = `
<div
  class="sc-bcXHqe kMOjqp"
  data-testid="track-info"
>
  <p
    class="sc-gswNZR hQCPLi"
    data-testid="track-name"
  >
    trackName
  </p>
  <p
    class="sc-dkrFOg"
    data-testid="track-artist"
  >
    trackArtist
  </p>
</div>
`;
